processor/PlanExecutor.go

```diff
@@ -16,6 +16,15 @@ import (
 )
 
 func merge(source map[string]interface{}, extra map[string]interface{}) map[string]interface{} {
+
+	if source == nil {
+		panic("Source map is nil")
```

Comment:
I am treating this one as exceptional as neither map should be nil so if they are something is wrong and it would be best to catch this at source.  If I return an error I am saying that it is possible for this to be nil and I will handle up the stack at runtime.  Happy to chat more about this one though.